src/Centreon/Infrastructure/Monitoring/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,430 @@
+
+    /**
+     * Initialized by the dependency injector.
+     *
+     * @param SqlRequestParametersTranslator $sqlRequestTranslator
+     */
+    public function setSqlRequestTranslator(SqlRequestParametersTranslator $sqlRequestTranslator): void
+    {
+        $this->sqlRequestTranslator = $sqlRequestTranslator;
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(
+                RequestParameters::CONCORDANCE_MODE_STRICT
+            );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function filterByAccessGroups(?array $accessGroups): TimelineRepositoryInterface
+    {
+        $this->accessGroups = $accessGroups;
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findTimelineEventsByHostAndService(int $hostId, int $serviceId = null): array
+    {
+        $timelineEvents = [];
+
+        if ($this->hasNotEnoughRightsToContinue()) {
+            return $timelineEvents;
+        }
+
+        $this->sqlRequestTranslator->setConcordanceArray([
+            'date' => 'event.timestamp'
+        ]);
+
+        $collector = new StatementCollector();
+        $request = $this->translateDbName('SELECT SQL_CALC_FOUND_ROWS '
+            . 'event.* '
+            . 'FROM (('
+            . $this->generateLogsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateCommentsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateAcknowledgementsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateDowntimesQuery($collector, $hostId, $serviceId)
+            . ')) AS  `event`');
+
+        // Search
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+
+        foreach ($this->sqlRequestTranslator->getSearchValues() as $key => $data) {
+            // @todo update when date parameter converter is available
+            $collector->addValue($key, strtotime(current($data)), key($data));
+        }
+        $request .= $searchRequest ? $searchRequest : '';
+
+        // Group
+        $request .= ' GROUP BY event.eventId';
+
+        // Sort
+        $request .= $this->sqlRequestTranslator->translateSortParameterToSql()
+            ?: ' ORDER BY event.timestamp DESC';
+
+        // Pagination
+        $request .= $this->sqlRequestTranslator->translatePaginationToSql();
+
+        $statement = $this->db->prepare($request);
+        $collector->bind($statement);
+
+        $statement->execute();
+
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        $this->sqlRequestTranslator->getRequestParameters()->setTotal(
+            (int)$result->fetchColumn()
+        );
+
+        while (false !== ($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            $newEvent = EntityCreator::createEntityByArray(
+                $this->detectClass($result['eventType']),
```

Comment:
I think this could be handled by a serializer discriminator. But let's keep it like that for now

Reply:
no it can't because the discriminator requires a specific field of the entity to detect which instance it is, in this case we have a class that has a subobject which changes implementations, I tried to do it with discriminator and it didnt work well